libraries/mbed/targets/hal/TARGET_ARM_SSG/TARGET_IOTSS/gpio_irq_api.c

```diff
@@ -0,0 +1,241 @@
+void gpio1_irq13(void) {handle_interrupt_in(29);}
+void gpio1_irq14(void) {handle_interrupt_in(30);}
+void gpio1_irq15(void) {handle_interrupt_in(31);}
+
+
+int gpio_irq_init(gpio_irq_t *obj, PinName pin, gpio_irq_handler handler, uint32_t id) {
+		if (pin == NC) {return -1;}
+		else {
+		
+		irq_handler = handler;
+		
+		int found_free_channel = 0;
+		int i = 0;
+		for (i=0; i<CHANNEL_NUM; i++) {
+				if (channel_ids[i] == 0) {
+						channel_ids[i] = id;
+						obj->ch = i;
+						found_free_channel = 1;
+						break;
+				}
+		}
+		if (!found_free_channel) return -1;
+		
+		
+		/* To select a pin for any of the eight pin interrupts, write the pin number
+		 * as 0 to 23 for pins PIO0_0 to PIO0_23 and 24 to 55.
+		 * @see: mbed_capi/PinNames.h
+		 */
+		if (pin <16)
+		{
+		CMSDK_GPIO_0->INTENSET |= (0x1 << pin);
+		}
+		
+		if (pin >= 16)
+		{
+		CMSDK_GPIO_1->INTENSET |= (0x1 << pin);
+		}
+
+		void (*channels_irq)(void) = NULL;
+		switch (obj->ch) {
+				case 0: channels_irq = &gpio0_irq0; break;
+				case 1: channels_irq = &gpio0_irq1; break;
+				case 2: channels_irq = &gpio0_irq2; break;
+				case 3: channels_irq = &gpio0_irq3; break;
+				case 4: channels_irq = &gpio0_irq4; break;
+				case 5: channels_irq = &gpio0_irq5; break;
+				case 6: channels_irq = &gpio0_irq6; break;
+				case 7: channels_irq = &gpio0_irq7; break;
+				case 8: channels_irq = &gpio0_irq8; break;
+				case 9: channels_irq = &gpio0_irq9; break;
+				case 10: channels_irq = &gpio0_irq10; break;
+				case 11: channels_irq = &gpio0_irq11; break;
+				case 12: channels_irq = &gpio0_irq12; break;
+				case 13: channels_irq = &gpio0_irq13; break;
+				case 14: channels_irq = &gpio0_irq14; break;
+				case 15: channels_irq = &gpio0_irq15; break;
+				case 16: channels_irq = &gpio1_irq0; break;
+				case 17: channels_irq = &gpio1_irq1; break;
+				case 18: channels_irq = &gpio1_irq2; break;
+				case 19: channels_irq = &gpio1_irq3; break;
+				case 20: channels_irq = &gpio1_irq4; break;
+				case 21: channels_irq = &gpio1_irq5; break;
+				case 22: channels_irq = &gpio1_irq6; break;
+				case 23: channels_irq = &gpio1_irq7; break;
+				case 24: channels_irq = &gpio1_irq8; break;
+				case 25: channels_irq = &gpio1_irq9; break;
+				case 26: channels_irq = &gpio1_irq10; break;
+				case 27: channels_irq = &gpio1_irq11; break;
+				case 28: channels_irq = &gpio1_irq12; break;
+				case 29: channels_irq = &gpio1_irq13; break;
+				case 30: channels_irq = &gpio1_irq14; break;
+				case 31: channels_irq = &gpio1_irq15; break;
+			
+		}
+		NVIC_SetVector((IRQn_Type)(PININT_IRQ + obj->ch), (uint32_t)channels_irq);
+		NVIC_EnableIRQ((IRQn_Type)(PININT_IRQ + obj->ch));
+	
+		return 0;
+	}
+}
+
+void gpio_irq_free(gpio_irq_t *obj) {
+}
+
+void gpio_irq_set(gpio_irq_t *obj, gpio_irq_event event, uint32_t enable) {
+		unsigned int ch_bit = (1 << obj->ch);
+		
+		// Clear interrupt
+		if (obj->ch <16)
+		{
+				if (!(CMSDK_GPIO_0->INTTYPESET & ch_bit))
```

Comment:
Please read, https://developer.mbed.org/teams/SDK-Development/wiki/mbed-sdk-coding-style and apply it to these files